model_server/src/model_server/model_types/base_model.py

```diff
@@ -0,0 +1,46 @@
+import asyncio
+from json import JSONDecodeError
+from starlette.responses import JSONResponse
+from starlette.routing import Route
+
+from model_server import settings
+
+
+def verify_fields_exist(*fields):
+    """A decorator that checks that the given `fields` are in the send JSON object."""
+    def verifier_wrapper(fn):
+        # It is assumed that this is going to be used to decorate a handler method, which has
+        # this signature
+        async def verify(self, payload):
+            missing = [field for field in fields if not payload.get(field)]
+            if missing:
+                return {'error': 'Missing fields in payload', 'missing': missing}, 400
+            return await fn(self, payload)
+        return verify
+    return verifier_wrapper
+
+
+def handler(fn):
```

Comment:
This is a decorator that will make sure that the request has a JSON object, extract it, and pass it to the wrapped function. So you can do something like:

```
class Bla:

    @handler
    def adder(payload):
        return payload.get('a') + payload.get('b')

def make_request(json):
    # create a proper request object with a `json()` function that will return the provided json
   
ble = Bla()
ble.adder(make_request(json={'a': 2, 'b': 4}))
```